File: frontend/demo/component/inputfields/input-field-label.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement, customElement } from 'lit-element';
import '@vaadin/vaadin-text-field/vaadin-email-field';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('input-field-label')
export class Example extends LitElement {
  constructor() {
    super();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(this.shadowRoot);
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-email-field label="Email address"></vaadin-email-field>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/inputfields/InputFieldLabel.java
package com.vaadin.demo.component.inputfields;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.EmailField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("input-field-label")
public class InputFieldLabel extends Div {

  public InputFieldLabel() {
    // tag::snippet[]
    EmailField emailField = new EmailField();
    emailField.setLabel("Email address");
    add(emailField);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<InputFieldLabel> { // hidden-source-line
  } // hidden-source-line
}
